src/api/emit.js

```diff
@@ -20,19 +31,48 @@ function createCustomEvent (name, opts) {
   return e;
 }
 
+function createReadableStopPropagation (oldStopPropagation) {
+  return function () {
+    this.isPropagationStopped = true;
+    oldStopPropagation.call(this);
+  };
+}
+
+function simulateBubbling (elem, cEvent) {
+  var status;
+  cEvent.stopPropagation = createReadableStopPropagation(cEvent.stopPropagation);
+  while (elem && !cEvent.isPropagationStopped) {
+    cEvent.currentTarget = elem;
+    if (elem.dispatchEvent(cEvent) === false) {
+      status = false;
+    }
+    elem = elem.parentNode;
+  }
+  return status;
+}
+
 function emitOne (elem, name, opts) {
+  var cEvent, shouldSimulateBubbling;
+
   /* jshint expr: true */
   opts.bubbles === undefined && (opts.bubbles = true);
   opts.cancelable === undefined && (opts.cancelable = true);
-  return elem.dispatchEvent(createCustomEvent(name, opts));
+  cEvent = createCustomEvent(name, opts);
+  shouldSimulateBubbling = opts.bubbles &&
+    !hasBubbleOnDetachedElements &&
+    !utilElementContains(document, elem);
+
+  return shouldSimulateBubbling ?
+    simulateBubbling(elem, cEvent) :
+    elem.dispatchEvent(cEvent);
 }
 
-export default maybeThis(function (elem, name, opts = {}) {
+export default function (elem, name, opts = {}) {
   var names = (typeof name === 'string' ? name.split(' ') : name);
   return names.reduce(function (prev, curr) {
-    if (!emitOne(elem, curr, opts)) {
+    if (emitOne(elem, curr, opts) === false) {
       prev.push(curr);
     }
     return prev;
   }, []);
```

Comment:
The intention of this reduction is to return the events whose default was not prevented?